typhon/cloudmask/aster.py

```diff
@@ -393,104 +415,77 @@ def retrieve_cloudmask(
+            probably_clear = ((r3N > 0.03) & (r5 > 0.01) & (0.7 < r3N2)
+                              & (r3N2 < 1.75) & (r12 < 1.45))
+            probably_cloudy = ((r3N > 0.03) & (r5 > 0.015) & (0.75 < r3N2)
+                               & (r3N2 < 1.75) & (r12 < 1.35))
+            confidently_cloudy = ((r3N > 0.065) & (r5 > 0.02) & (0.8 < r3N2)
+                                 & (r3N2 < 1.75) & (r12 < 1.2))
+        
+        mask_swathedge = (np.isnan(r1) | np.isnan(r2) | np.isnan(r3N) | np.isnan(r5))
+        
+        clmask[probably_clear] = PROBABLY_CLEAR
+        clmask[probably_cloudy] = PROBABLY_CLOUDY
+        clmask[confidently_cloudy] = CONFIDENTLY_CLOUDY
+        clmask[mask_swathedge] = np.nan
 
         if include_thermal_test:
             ### TEST 5 ###
             # Uncertain warm ocean pixels, higher than the 5th percentile of
             # brightness temperature values from all "confidently clear"
             # labeled pixels, are overwritten with "confidently clear".
+            # Read thermal (TIR) channel at 90m resolution and match VNIR
+            # resolution.
+            bt14 = self.get_brightnesstemperature(channel="14")
+            bt14 = np.repeat(np.repeat(bt14, 6, axis=0), 6, axis=1)
 
             # Check for available "confidently clear" pixels.
-            nc = np.sum(clmask == 2) / np.sum(~np.isnan(clmask))
+            nc = np.sum(clmask == CONFIDENTLY_CLEAR) / np.sum(~np.isnan(clmask))
             if nc > 0.03:
-                bt14_p05 = np.nanpercentile(bt14[clmask == 2], 5)
+                bt14_p05 = np.nanpercentile(bt14[clmask == CONFIDENTLY_CLEAR], 5)
             else:
                 # If less than 3% of pixels are "confidently clear", test 5
                 # cannot be applied according to Werner et al., 2016. However,
                 # a sensitivity study showed that combining "probably clear"
                 # and "confidently clear" pixels in such cases leads to
                 # plausible results and we derive a threshold correspondingly.
                 bt14_p05 = np.nanpercentile(
-                    bt14[np.logical_or(clmask == 2, clmask == 3)], 5
-                )
+                    bt14[(clmask == CONFIDENTLY_CLEAR) | (clmask == PROBABLY_CLEAR)], 5)
 
             with np.warnings.catch_warnings():
                 np.warnings.filterwarnings("ignore", r"invalid value encountered")
                 # Pixels with brightness temperature values above the 5th
                 # percentile of clear ocean pixels are overwritten with
                 # "confidently clear".
-                clmask[np.logical_and(bt14 > bt14_p05, ~np.isnan(clmask))] = 2
+                clmask[(bt14 > bt14_p05) & ~np.isnan(clmask)] = CONFIDENTLY_CLEAR
 
-            # Combine swath edge pixels.
-            clmask[np.logical_or(np.isnan(clmask), np.isnan(bt14))] = np.nan
+            # Add swath edge pixels of thermal channel.
+            clmask[np.isnan(bt14)] = np.nan
```

Comment:
What do you mean with "add"? Does not the line just copy the NaN-mask form the thermal channel?

Reply:
The thermal channel has the lowest resolution and thus can overwrite pixel measurements at the swath edge from the higher resolved channels. I'll change it to say "apply" instead of "add".